File: Leetcode/python/sol1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen:dict = {}
        for idx, val in enumerate(nums):
            need:int = target - val
            if need in seen:
                return [seen[need], idx]
            else:
                seen[val] = idx

        return []

File: Training/sol1.cpp
//https://leetcode.com/problems/two-sum/description/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        for (int i = 0; i < nums.size(); i++){
            int needs = target - nums[i];
            auto end = seen.find(needs);
            if(end != seen.end()){
                return {i, seen[needs]};
            }
            seen.insert({nums[i], i});
        }
        return {};
    }
};